qcodes/tests/test_threading.py

```diff
@@ -0,0 +1,76 @@
+"""
+Test suite for utils.threading.*
+"""
+import pytest
+from typing import Any
+import threading
+
+from qcodes.instrument.parameter import Parameter, ParamRawDataType
+from qcodes.utils.threading import call_params_threaded
+
+from .instrument_mocks import DummyInstrument
+
+
+class ParameterWithThreadKnowledge(Parameter):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    def get_raw(self) -> ParamRawDataType:
+        return threading.get_ident()
+
+
+@pytest.fixture(name='dummy_1', scope='function')
+def _dummy_dac1():
+    instrument = DummyInstrument(
+        name='dummy_1', gates=['ch1'])
+
+    instrument.add_parameter(name='voltage_1',
+                             parameter_class=ParameterWithThreadKnowledge)
+
+    instrument.add_parameter(name='voltage_2',
+                             parameter_class=ParameterWithThreadKnowledge)
+
+    try:
+        yield instrument
+    finally:
+        instrument.close()
+
+
+@pytest.fixture(name='dummy_2', scope='function')
+def _dummy_dac2():
+    instrument = DummyInstrument(
+        name='dummy_2', gates=['ch1'])
+
+    instrument.add_parameter(name='voltage_1',
+                             parameter_class=ParameterWithThreadKnowledge)
+
+    instrument.add_parameter(name='voltage_2',
+                             parameter_class=ParameterWithThreadKnowledge)
+
+    try:
+        yield instrument
+    finally:
+        instrument.close()
+
+
+def test_call_params_threaded(dummy_1, dummy_2):
+
+    params_output = call_params_threaded((dummy_1.voltage_1,
+                                          dummy_1.voltage_2,
+                                          dummy_2.voltage_1,
+                                          dummy_2.voltage_2))
+
+    param1 = params_output[0][0]
+    thread_id1 = params_output[0][1]
+    for i in range(1, 4):
+        param2 = params_output[i][0]
+        thread_id2 = params_output[i][1]
+
+        if param1.instrument is param2.instrument:
+            assert thread_id1 == thread_id2
+
+        if param1.instrument is not param2.instrument:
+            assert thread_id1 != thread_id2
+
+        param1 = param2
+        thread_id1 = thread_id2
```

Comment:
@Akshita07 i am trying to see if this assertion algorithm always works, and i'm having trouble with processing it. it might be just me, but for the sake of providing an alternative i ended up with the following implementation - do you think it's clearer and more strict and explicit? 
```suggestion
    params_per_thread_id = defaultdict(set)
    for param, thread_id in params_output:
        params_per_thread_id[thread_id].add(param)
    assert len(params_per_thread_id) == 2
    expected_params_per_thread = {
        {dummy_1.voltage_1, dummy_1.voltage_2},
        {dummy_2.voltage_1, dummy_2.voltage_2}
    }
    assert set(params_per_thread_id.values()) == expected_params_per_thread
```

Reply:
@astafan8 it's an interesting use of defaultdict. I'll note it down for future implementation purposes. And yes, your implementation is much more clearer, strict and explicit. I am comiting this as well :)